### ISSUE TITLE
Enhancement: Show console error output on :19080 monitor page.

### ISSUE THREAD

**Issue by qrli** [Azure/service-fabric-issues#250]
<SetupEntryPoint> and <EntryPoint> may encounter some error which cannot be logged by application. They are only captured by Service Fabric and written to console logs if enabled. While we can remote desktop to the VM and find the log, it is really time consuming on slow connection, only to find small errors. This is especially annoying when you have many teams deploying to the same cluster; many people need to remote to the VMs.

It is far better if the monitor page simply show the reason (some piece of the std err output), other than only process return code. Or provide a download link for the console logs.

**Comment by MikkelHegn** [comment]
@qrli: seems some of your feedback was cut in editing?

**Comment by qrli** [comment]
@MikkelHegn Not caused by me, but github. It seems github starts to filter out &lt;...&gt;...

**Comment by masnider** [comment]
@qrli Most of these should be showing up as health messages, which are visible in SFX. If they're not, can you give us an example of such an error and the log that helped you figure it out?

**Comment by qrli** [comment]
@masnider Thanks for looking at this.
The health messages include a good amount of info, but not for **guest executable** nor **setup entry point script**. I can only see there that an entry point or setup entry point failed, but I don't see **why**.

E.g.: I have a setup entry point script, which works locally but failed on Azure. One common cause for us is that we have several clusters created at different times with different versions of ARM and VM setup script, so the environments may be slightly different. Then, we have to remote desktop (need asking for permission to access) to the cluster and look at the captured **console error logs**. It may be missing dependencies, incorrect/expired certificates, conflicts with another application, etc.

**Comment by MikkelHegn** [comment]
@qrli: @cwe1ss wrote a good SO response to this here: https://stackoverflow.com/questions/37885213/azure-service-fabric-activation-error

**Comment by qrli** [comment]
@MikkelHegn That's not a solution but more of an example of this issue. RDP to the VM to see the log is what exactly we want to avoid.
When you have many teams, you will sure want to control the access to the VMs, to avoid somebody messing up the environment. But for now, each team has to RDP to the VMs only to view the console error logs. It will be much easier if the health message included the console error output instead of just "entry point failed to start".

**Comment by MikkelHegn** [comment]
@qrli I agree - the intention was to give a pointer to help you with the issue at hand. We've know this is an enhancement we need to make to the product / tooling around SF.
